eta-meeting-organizer-backend/src/main/java/hu/flowacademy/meetingorganizer/service/MeetingRoomService.java

```diff
@@ -0,0 +1,46 @@
+package hu.flowacademy.meetingorganizer.service;
+
+import hu.flowacademy.meetingorganizer.persistance.model.MeetingRoom;
+import hu.flowacademy.meetingorganizer.persistance.repository.MeetingRoomRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class MeetingRoomService {
+
+    private MeetingRoomRepository meetingRoomRepository;
+
+    public List<MeetingRoom> findAllMeetingRooms () {
+        return meetingRoomRepository.findAll();
+    }
+
+    public MeetingRoom findOneMeetingRoomById (Long id) {
+        return meetingRoomRepository.findById(id).orElseThrow(()-> new ResponseStatusException(HttpStatus.NOT_FOUND));
+    }
+
+    public MeetingRoom createMeetingRoom (MeetingRoom meetingRoom) {
+        return meetingRoomRepository.save(meetingRoom);
+    }
+
+    public void deleteMeetingRoom (Long id) {
+        meetingRoomRepository.deleteById(id);
+    }
+
+    public MeetingRoom updateMeetingRoom (Long id, MeetingRoom meetingRoom) {
+        if (meetingRoomRepository.findById(id).isPresent()) {
```

Comment:
Használjuk ki az Optional beépített metódusait: `Optional.orElseThrow(() -> new ResponseStatusException(...))`, és utána már if nélkül folytathatjuk tovább a business logic-unkat. (~ early return)

Reply:
Ez működik-e?
```java 
meetingRoom.setId(id); 
return meetingRoomRepository.save(meetingRoom);
```
Ha nem működik, akkor kérjük le a db-ből a meetingRoom objectet (ahogy eddig), és updateljük a fieldjeit (azokra, amiket paraméterben kaptunk) és utána mentsünk.

Reply:
Például:
```java
    MeetingRoom room = meetingRoomRepository.findById(id).orElseThrow(NoSuchElementException::new);
        room.setId(id);
        room.setName(meetingRoom.getName());
        room.setNumberOfSeats(meetingRoom.getNumberOfSeats());
        return meetingRoomRepository.save(room);
```